Databases/VSSTester/VSSTester.ps1

```diff
@@ -1,33 +1,81 @@
-#               execute VSS backup using DiskShadow,
-#               delete the VSS snapshot post-backup
-# - stop PowerShell transcript
-#
-#################################################################################
-[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingEmptyCatchBlock', '', Justification = 'Allowing empty catch blocks for now as we need to be able to handle the exceptions.')]
-[Diagnostics.CodeAnalysis.SuppressMessageAttribute('CustomRules\AvoidUsingReadHost', '', Justification = 'Do not want to change logic of script as of now')]
+<#
+.SYNOPSIS
+    Test and/or trace VSS functionality on Exchange Server.
+.DESCRIPTION
+    Test and/or trace VSS functionality on Exchange Server.
+.LINK
+    https://microsoft.github.io/CSS-Exchange/Databases/VSSTester/
+.EXAMPLE
+    .\VSSTester -TraceOnly -DatabaseName "Mailbox Database 1637196748"
+    Enables tracing of the specified database. The user may then attempt a backup of that database
+    and use Ctrl-C to stop data collection after the backup attempt completes.
+.EXAMPLE
+    .\VSSTester -DiskShadow -DatabaseName "Mailbox Database 1637196748" -DatabaseDriveLetter M -LogDriveLetter N
+    Enables tracing and then uses DiskShadow to snapshot the specified database. If the database and logs
+    are on the same drive, the snapshot is exposed as M: drive. If they are on separate drives, the snapshots are
+    exposed as M: and N:. The user is prompted to stop data collection and should typically wait until
+    log truncation has occurred before doing so, so that the truncation is traced.
+.EXAMPLE
+    .\VSSTester -WaitForWriterFailure -DatabaseName "Mailbox Database 1637196748"
+    Enables circular tracing of the specified database, and then polls "vssadmin list writers" once
+    per minute. When the writer is no longer present, indicating a failure, tracing is stopped
+    automatically.
+#>
 [CmdletBinding()]
 param(
+    # Enable tracing and wait for the user to run a third-party backup solution.
+    [Parameter(Mandatory = $true, ParameterSetName = "TraceOnly")]
+    [switch]
+    $TraceOnly,
+
+    # Enable tracing and perform a database snapshot with DiskShadow.
+    [Parameter(Mandatory = $true, ParameterSetName = "DiskShadow")]
+    [switch]
+    $DiskShadow,
+
+    # Enable tracing and automatically stop when the Microsoft Exchange Writer fails.
+    [Parameter(Mandatory = $true, ParameterSetName = "WaitForWriterFailure")]
+    [switch]
+    $WaitForWriterFailure,
+
+    # Name of the database to focus tracing on.
+    [Parameter(Mandatory = $true, ParameterSetName = "TraceOnly")]
+    [Parameter(Mandatory = $true, ParameterSetName = "DiskShadow")]
+    [Parameter(Mandatory = $true, ParameterSetName = "WaitForWriterFailure")]
+    [string]
+    $DatabaseName,
+
+    # Drive letter on which to expose the database snapshot.
+    [Parameter(Mandatory = $true, ParameterSetName = "DiskShadow")]
+    [string]
+    $DatabaseDriveLetter,
```

Comment:
Haven't done any testing, but is it a concern if `M:` or `M:\` is provided?

Reply:
Not anymore. Added [ValidateLength] attribute.

![image](https://github.com/microsoft/CSS-Exchange/assets/4518572/ce162f00-dec7-40fe-9d35-e50064aeda2b)